README.rst

```diff
@@ -15,104 +15,171 @@ sqlitedict -- persistent ``dict``, backed-up by SQLite and pickle
-  ...     # no explicit commit here
-  >>> with SqliteDict('./my_db.sqlite') as mydict:  # re-open the same DB
-  ...     print mydict['some_key']  # outputs 'first value', not 'new value'
+    from sqlitedict import SqliteDict
+    db = SqliteDict("./db.sqlite")
 
+    print("There are {} items in the database".format(len(db)))
 
-Features
---------
+    # standard dict interface. items() values() keys() etc...
+    for key, item in db.items():
+        print("{}={}".format(key, item))
 
-* Values can be **any picklable objects** (uses ``cPickle`` with the highest protocol).
-* Support for **multiple tables** (=dicts) living in the same database file.
-* Support for **access from multiple threads** to the same connection (needed by e.g. Pyro).
-  Vanilla sqlite3 gives you ``ProgrammingError: SQLite objects created in a thread can
-  only be used in that same thread.``
+    # note that this is implemented efficiently.
+    # len() is calling sqlite to count items, it doesn't load objects in memory.
+    # in() is iterating on the objects one by one.
 
-  Concurrent requests are still serialized internally, so this "multithreaded support"
-  **doesn't** give you any performance benefits. It is a work-around for sqlite limitations in Python.
+Context Manager
+---------------
 
-* Support for **custom serialization or compression**:
+.. code-block:: python
 
-  .. code-block:: python
+    from sqlitedict import SqliteDict
 
-      # use JSON instead of pickle
-      >>> import json
-      >>> mydict = SqliteDict('./my_db.sqlite', encode=json.dumps, decode=json.loads)
+    # the database is automatically closed when leaving the with section.
+    # uncommited objects are not saved on close. REMEMBER TO COMMIT!
 
-      # apply zlib compression after pickling
-      >>> import zlib, pickle, sqlite3
-      >>> def my_encode(obj):
-      ...     return sqlite3.Binary(zlib.compress(pickle.dumps(obj, pickle.HIGHEST_PROTOCOL)))
-      >>> def my_decode(obj):
-      ...     return pickle.loads(zlib.decompress(bytes(obj)))
-      >>> mydict = SqliteDict('./my_db.sqlite', encode=my_encode, decode=my_decode)
+    with SqliteDict("./db.sqlite") as db:
+        print("There are {} items in the database".format(len(db)))
 
 
-Installation
-------------
+Tables
+------
 
-The module has no dependencies beyond Python itself. The minimum Python version is 2.5, continuously tested on Python 2.6, 2.7, 3.3 and 3.4 `on Travis <https://travis-ci.org/RaRe-Technologies/sqlitedict>`_.
+A default table is used when no table name is specified.
 
-Install or upgrade with::
+.. code-block:: python
 
-    pip install -U sqlitedict
+    from sqlitedict import SqliteDict
 
-or from the `source tar.gz <http://pypi.python.org/pypi/sqlitedict>`_::
+    # support for multiple tables in the same files.
+    # writes are serialized. consider splitting large databases (> 1 GB) for performance.
+    products      =  SqliteDict("./db.sqlite", tablename="product", autocommit=True)
```

Comment:
PEP8, here and everywhere.

```suggestion
    products =  SqliteDict("./db.sqlite", tablename="product", autocommit=True)
```

Reply:
the assignment is aligned with the assignment on next line for readability, it's not against PEP8. Well, I can't edit if you really care about that?

Reply:
Yes please.